typescript/src/schema/proto.ts

```diff
@@ -288,6 +289,16 @@ function toLRInterface(type: string, inputType: string) {
   return inputType.replace(/\.([^.]+)$/, '.I' + type);
 }
 
+export function getHeaderParms(rule: plugin.google.api.IHttpRule): string[] {
+  const message =
+    rule.post || rule.delete || rule.get || rule.put || rule.patch;
+  if (message) {
+    const res = message.match(/{(.*?)(?==)/);
```

Comment:
No need to have `(?=  )` around the final `=`. The meaning of `(?=  )` is "group together but don't extract to a match group". Since you only have one character `=` inside `(?=  )`, there is no need to put it into any kind of brackets if you don't need to extract them.

Just `/{(.*?)=/` should work.

Reply:
Cool, just realize that for `/v1/{crypto_key_version.name=projects/*/locations/*/keyRings/*/cryptoKeys/*/cryptoKeyVersions/`

`/{(.*?)=/` will match `[{crypto_key_version.name=, crypto_key_version.name]`
and `/{(.*?)(?==)/` will match `[{crypto_key_version.name, crypto_key_version.name]` and I and taking the first element which will give the wrong answer of `crypto_key_version.name=`.

But with choosing the second element, it works perfect